### DISCUSSION TITLE
How to handle buffer data between processors chain

### DISCUSSION THREAD

**Discussion by licongchao** [redpanda-data/connect#2203] [General]
Benthos is fantastic. I'm looking into it to solve some data processing requirements.
The requirements is to collect data from external service and do some data processing. i already added a customized input and working on the customized processor. The processor output should be buffered when specific logic/condition meets (**Wait** for other data to arrive and do **periodical** timer to deliver output result).
Is this logic suitable to be placed in the processor?

**Comment by Jeffail** [comment]
Hey @licongchao, there are buffer plugins which sound like they suit your use case better: https://pkg.go.dev/github.com/benthosdev/benthos/v4@v4.23.0/public/service#Environment.RegisterBatchBuffer

Converting to a discussion as per https://github.com/benthosdev/benthos/issues/2026

**Comment by licongchao** [comment]
Thanks for quick response. It's a genius design. The docs of RegiserXXX inspired me. i can also add external channel into Benthos if there is no need to reuse this component.
